File: ContainsDuplicate/containsDuplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        """
        data = set()
        for i in nums:
            if i in data:
                return True
            data.add(i)
        return False
        """
        #new solution
        dupMap = {}  # value -> count

        for x in nums:
            if x not in dupMap:
                dupMap[x] = 1
            else:
                return True

File: ContainsDuplicate/containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        //iterate thru, add to map
        //if not in map, continue
        //if in map, return true as contains duplicate
        //else return false
        Set<Integer> set = new HashSet<Integer>();
        boolean result = false;
        
        for(int i = 0; i < nums.length; i++){
            if(map.contains(nums[i])){
                result = true;
            } else {
                map.add(nums[i]);
            }
        }
        return result;
    }
}
